langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingResponseBuilder.java

```diff
@@ -134,19 +174,28 @@ private TokenUsage tokenUsage(String content, Tokenizer tokenizer) {
         return new TokenUsage(inputTokenCount, outputTokenCount);
     }
 
-    private TokenUsage tokenUsage(ToolExecutionRequest toolExecutionRequest, Tokenizer tokenizer, boolean forcefulToolExecution) {
+    private TokenUsage tokenUsage(List<ToolExecutionRequest> toolExecutionRequests, Tokenizer tokenizer, boolean forcefulToolExecution) {
         if (tokenizer == null) {
             return null;
         }
 
         int outputTokenCount = 0;
         if (forcefulToolExecution) {
             // OpenAI calculates output tokens differently when tool is executed forcefully
-            outputTokenCount += tokenizer.estimateTokenCountInForcefulToolExecutionRequest(toolExecutionRequest);
+            for (ToolExecutionRequest toolExecutionRequest : toolExecutionRequests) {
+                outputTokenCount += tokenizer.estimateTokenCountInForcefulToolExecutionRequest(toolExecutionRequest);
+            }
         } else {
-            outputTokenCount = tokenizer.estimateTokenCountInToolExecutionRequests(singletonList(toolExecutionRequest));
+            outputTokenCount = tokenizer.estimateTokenCountInToolExecutionRequests(toolExecutionRequests);
         }
 
         return new TokenUsage(inputTokenCount, outputTokenCount);
     }
+
+    private static class ToolExecutionRequestBuilder {
+
+        private final StringBuffer idBuilder = new StringBuffer();
+        private final StringBuffer nameBuilder = new StringBuffer();
+        private final StringBuffer argumentsBuilder = new StringBuffer();
```

Comment:
Can ignore if wanted, is the expectation for these to be used in a threaded scenario? If not, `StringBuilder` is preferable as it doesn't have synchronization as it's not thread safe.